nuxeo-retention/src/main/java/org/nuxeo/retention/service/RetentionManagerImpl.java

```diff
@@ -185,6 +197,34 @@ protected void checkCanAttachRule(DocumentModel document, RetentionRule rule, Co
         }
     }
 
+    @Override
+    public void fireRetentionEvent(String eventName, String eventInput, boolean audit, CoreSession session) {
+        NuxeoPrincipal principal = session.getPrincipal();
+        if (!principal.isAdministrator() && !principal.isMemberOf(RECORD_MANAGER_GROUP_NAME)) {
+            throw new NuxeoException(
+                    String.format("User: %s is not authorized to fire retention event", principal.getPrincipalId()),
```

Comment:
`principal.getName()` might be more relevant, aka username instead of generated id